docs/introduction.rst

```diff
@@ -303,5 +303,28 @@ Cons
 
 
 Saliency Algorithms
---------------------------------
-*Discuss the provided xaitk-saliency algorithms in terms of the above.*
+-------------------
+
+The xaitk-saliency package provides two types of black-box algorithms to create explanations: **image perturbation
+algorithms** that generate perturbed images and **saliency algorithms** that generate saliency heatmaps based on the
+black-box observations of the resulting perturbed images.
+
+The saliency algorithms can be organized into three classes according to their respective tasks:
+
+* **Image classification** - Randomized Input Sampling for Explanation (RISE) and RISEScoring algorithms, and the
+  OcclusionScoring algorithm
+
+* **Image similarity** - the Similarity Based Saliency Maps (SBSM) algorithm
+
+* **Object detection** - D-RISE and D-RISEScoring algorithms
+
+In addition, there are stacked (end-to-end) implementations for image classification that use multiple algorithms:
+
+*  RISEStack (which uses the RISEGrid perturbation and the RISEScoring saliency algorithms), and
+*  SlidingWindowStack (which uses the SlidingWindows perturbation and the OcclusionScoring saliency
+   algorithms).
+
+.. note:: Stacked implementations are planned in the future for the image similarity and object detection task domains.
+
+The sections that follow describe the interfaces available for perturbing images and generating the saliency heatmaps,
+and the implementations for which the algorithms are candidates.
```

Comment:
I think this is a good step in the right direction. My sense is that we only need to introduce at a high level the different saliency algorithms, without necessarily referencing to their exact implementation in the package. I imagine that this section is something a new user would visit to better understand what specific algorithms are available to them and how they map to the existing XAI literature.

I've made some suggested changes below (newlines have to fixed and open to discussion):

```suggestion
The xaitk-saliency package currently provides several black-box XAI algorithms. These algorithms follow a general pattern composed of two sequential steps: **image perturbation** followed by **heatmap generation**. **Image perturbation** involves generating perturbed versions of the input image by applying a set of perturbation masks. **Heatmap generation** involves generating saliency heatmaps based on how the black-box model outputs change as a result of image perturbation. This technical design choice allows for modularization of the **image perturbation** and **heatmap generation** components of the algorithm. Additionally, by formulating the algorithms in this manner, the exact operation of the black-box model is not needed by the algorithm and is flexible to be determined by the user.

The saliency algorithms can also be roughly organized according to their respective tasks:

TODO: Maybe make this into a table?
* **Image classification** - Occlusion-based Saliency [1] and Randomized Input Sampling for Explanation (RISE) [2]

* **Image similarity** - Similarity Based Saliency Maps (SBSM) algorithm [3]

* **Object detection** - Detector-RISE (D-RISE) [4]

TODO: Add references here
[1]
[2]
[3]
[4]
```

Reply:
Yes this looks good to me, thanks, other than the table (I think that'd be overkill with just 4 algorithms).

Reply:
One clarification needed @brianhhu: 

> by formulating the algorithms in this manner, the exact operation of the black-box model is not needed by the algorithm and is flexible to be determined by the user.

are you saying the model is flexible and can accept an algorithm chosen by the use?

Reply:
Maybe this is somewhat clearer?

> Additionally, by formulating the algorithms in this manner, knowledge of the exact operation of the black-box model is not needed by the algorithm and can be flexibly determined by the user.

The main point is that the black-box model being used can be anything from the perspective of the XAI algorithm, i.e. 1) the XAI algorithm only cares about the inputs/outputs of this black box, and 2) the black-box algorithm is something the user is free to choose and configure in any way according to the problem domain. There may be some confusion with how we use "algorithm", and whether it refers to the XAI component or the underlying black-box model.